klippy/extras/lis2dw.py

```diff
@@ -0,0 +1,269 @@
+
+                raw_xyz = (rx, ry, rz)
+
+                x = round(raw_xyz[x_pos] * x_scale, 6)
+                y = round(raw_xyz[y_pos] * y_scale, 6)
+                z = round(raw_xyz[z_pos] * z_scale, 6)
+
+                ptime = round(time_base + (msg_cdiff + i) * inv_freq, 6)
+                samples[count] = (ptime, x, y, z)
+                count += 1
+        self.clock_sync.set_last_chip_clock(seq * SAMPLES_PER_BLOCK + i)
+        del samples[count:]
+        return samples
+    def _update_clock(self, minclock=0):
+        # Query current state
+        for retry in range(5):
+            params = self.query_lis2dw_status_cmd.send([self.oid],
+                                                        minclock=minclock)
+            fifo = params['fifo'] & 0x1f
+            if fifo <= 32:
+                break
+        else:
+            raise self.printer.command_error("Unable to query lis2dw fifo")
+        mcu_clock = self.mcu.clock32_to_clock64(params['clock'])
+        sequence = (self.last_sequence & ~0xffff) | params['next_sequence']
+        if sequence < self.last_sequence:
+            sequence += 0x10000
+        self.last_sequence = sequence
+        buffered = params['buffered']
+        limit_count = (self.last_limit_count & ~0xffff) | params['limit_count']
+        if limit_count < self.last_limit_count:
+            limit_count += 0x10000
+        self.last_limit_count = limit_count
+        duration = params['query_ticks']
+        if duration > self.max_query_duration:
+            # Skip measurement as a high query time could skew clock tracking
+            self.max_query_duration = max(2 * self.max_query_duration,
+                                          self.mcu.seconds_to_clock(.000005))
+            return
+        self.max_query_duration = 2 * duration
+        msg_count = (sequence * SAMPLES_PER_BLOCK
+                     + buffered // BYTES_PER_SAMPLE + fifo)
+        # The "chip clock" is the message counter plus .5 for average
+        # inaccuracy of query responses and plus .5 for assumed offset
+        # of lis2dw hw processing time.
+        chip_clock = msg_count + 1
+        self.clock_sync.update(mcu_clock + duration // 2, chip_clock)
+    def _start_measurements(self):
+        if self.is_measuring():
+            return
+        # In case of miswiring, testing LIS2DW device ID prevents treating
+        # noise or wrong signal as a correctly initialized device
+        dev_id = self.read_reg(REG_LIS2DW_WHO_AM_I_ADDR)
+        logging.info("lis2dw_dev_id: %x", dev_id)
+        if dev_id != LIS2DW_DEV_ID:
+            raise self.printer.command_error(
+                "Invalid lis2dw id (got %x vs %x).\n"
+                "This is generally indicative of connection problems\n"
+                "(e.g. faulty wiring) or a faulty lis2dw chip."
+                % (dev_id, LIS2DW_DEV_ID))
+        # Setup chip in requested query rate
+        # ODR/2 +-16g
```

Comment:
You could also mention that you enable low-noise mode here.

Reply:
Hi, 0x30 is diasble low-noise mode. Do I need to set it to 0x34?
![image](https://github.com/Klipper3d/klipper/assets/38851044/fd5908a7-9b4a-4fd5-bc30-0157ac84ea89)

Reply:
Ah, yes, you are right, 0x34 enables the low-noise configuration. Well, I did not test this accelerometer, but just looking at the datasheet, I see no drawbacks of enabling it. It just makes the sensor consume ~30% more current, but it is still little, and if the powerchain is adequate, it should cause no problems.

Reply:
enabled now